### PR TITLE
Updated so block rate is still calculated when interval is timestamp

### PR DESCRIPTION
**Pull request by steegecs**

### CODE REVIEW

File: subgraphs/_reference_/src/common/rewards.ts [hunk 1]
```diff
@@ -193,5 +194,7 @@ function getStartingBlockRate(): BigDecimal {
     // else if (dataSource.network() == "harmony") return BigDecimal.fromString("13.39")
     // else if (dataSource.network() == SubgraphNetwork.MOONBEAM) return BigDecimal.fromString("13.39")
     // else if (dataSource.network() == SubgraphNetwork.MOONRIVER) return BigDecimal.fromString("13.39")
-    else return BIGDECIMAL_ZERO
+    else {
+        log.warning("getStartingBlockRate(): Network not found", [])
+        return BIGDECIMAL_ZERO
```
Comment (this-username-is-taken) [inline subgraphs/_reference_/src/common/rewards.ts:199]: no closing bracket?

**Comment by this-username-is-taken** [comment]
You should always do
```
if () {
} else {
}
```
for readability

**Comment by this-username-is-taken** [comment]
Probably best to use prettier to format everything

**Comment by this-username-is-taken** [comment]
Also remove the `.DS_Store` files

**Comment by steegecs** [comment]
> You should always do
> 
> ```
> if () {
> } else {
> }
> ```
> 
> for readability

So you want me to update the one line conditional statements that do not use brackets? Or just the if/else in the while loop?

**Comment by this-username-is-taken** [comment]
> So you want me to update the one line conditional statements that do not use brackets? Or just the if/else in the while loop?

every where, for example, change:

``` javascript
if (windowIndexBlockTimestamp < startTimestamp) {
    circularBuffer.windowStartIndex = circularBuffer.windowStartIndex + INT_TWO
    if (circularBuffer.windowStartIndex >= circularBuffer.bufferSize) circularBuffer.windowStartIndex = INT_ZERO
} else break
```
to
``` javascript
if (windowIndexBlockTimestamp < startTimestamp) {
    circularBuffer.windowStartIndex = circularBuffer.windowStartIndex + INT_TWO
    if (circularBuffer.windowStartIndex >= circularBuffer.bufferSize) {
        circularBuffer.windowStartIndex = INT_ZERO
    }
} else {
    break
}
```

**Comment by steegecs** [comment]
> > So you want me to update the one line conditional statements that do not use brackets? Or just the if/else in the while loop?
> 
> every where, for example, change:
> 
> ```js
> if (windowIndexBlockTimestamp < startTimestamp) {
>     circularBuffer.windowStartIndex = circularBuffer.windowStartIndex + INT_TWO
>     if (circularBuffer.windowStartIndex >= circularBuffer.bufferSize) circularBuffer.windowStartIndex = INT_ZERO
> } else break
> ```
> 
> to
> 
> ```js
> if (windowIndexBlockTimestamp < startTimestamp) {
>     circularBuffer.windowStartIndex = circularBuffer.windowStartIndex + INT_TWO
>     if (circularBuffer.windowStartIndex >= circularBuffer.bufferSize) {
>         circularBuffer.windowStartIndex = INT_ZERO
>     }
> } else {
>     break
> }
> ```

Okay! I fixed this

File: subgraphs/_reference_/src/common/rewards.ts [hunk 2]
```diff
@@ -42,156 +49,207 @@ export const BUFFER_SIZE = 144
...
-    // The timestamp at the start of the window (default 24 hours in seconds).
-    let startTimestamp = currentTimestampI32 - WINDOW_SIZE_SECONDS
-
-    // Make sure to still have 2 blocks to calculate rate (This shouldn't happen past the beginning).
-    while(abs(circularBuffer.nextIndex - circularBuffer.windowStartIndex) > INT_FOUR) {
-        let windowIndexBlockTimestamp = blocks[circularBuffer.windowStartIndex]
+  // Add current timestamp and block numnber to array if new block is at least X blocks later than previously stored.
+  // Used to save memory and efficiency on array resizing.
+  let recentSavedTimestamp: i32;
+  if (circularBuffer.nextIndex == INT_ZERO)
```
Comment (this-username-is-taken) [inline subgraphs/_reference_/src/common/rewards.ts:126]: braces

File: subgraphs/_reference_/src/common/rewards.ts [hunk 3]
```diff
@@ -42,156 +49,207 @@ export const BUFFER_SIZE = 144
...
+
+  // Update BlockTracker with new values.
+  circularBuffer.blocksPerDay = normalizedBlockSpeed;
+  circularBuffer.blocks = blocks;
+
+  circularBuffer.save();
+
+  if (rewardType == RewardIntervalType.TIMESTAMP) {
+    return rewardRate.times(RATE_IN_SECONDS_BD);
+  } else return rewardRate.times(circularBuffer.blocksPerDay);
```
Comment (this-username-is-taken) [inline subgraphs/_reference_/src/common/rewards.ts:195]: braces

**Pull request merged**